m2701drv/m2701drv/config.pl.org
#Configuration of m2xxxdrv Client.

if(
$::NodeName eq 'm2701drv'
){###############################################################################

#STARS server
$::Server = 'localhost';

## For NPORT interface
$::NPORT_HOST  = '192.168.156.51'; #NPort host name.
$::NPORT_PORT  = 1394;             #NPort port number.

$::DESTINATION{'ch01'} = "101";
$::DESTINATION{'ch02'} = "102";
$::DESTINATION{'ch03'} = "103";
$::DESTINATION{'ch04'} = "104";
$::DESTINATION{'ch05'} = "105";
$::ALIAS{'all'} = "ch01,ch02,ch03,ch04,ch05";


$::NPLC_MAX = 50;

$::RegFile = 'reg.txt';		#Registry fileM

$::DMMTYPE_M2000 = 0;

#functions----------------------------------------
%::dmm_func_functions = (
'ACI',   'CURR:AC',
'DCI',   'CURR:DC',
'ACV',   'VOLT:AC',
'DCV',   'VOLT:DC',
'2OHM',  'RES',
'4OHM',  'FRES',
'FREQ',  'FREQ',
'TEMP',  'TEMP'
);

%::dmm_NPLC_functions = (
'ACI',   ':CURR:AC:NPLC',
'DCI',   ':CURR:DC:NPLC',
'ACV',   ':VOLT:AC:NPLC',
'DCV',   ':VOLT:DC:NPLC',
'2OHM',  ':RES:NPLC',
'4OHM',  ':FRES:NPLC',
'TEMP',  ':TEMP:NPLC',
);

%::dmm_range_functions = (
'ACI',   ':CURR:AC:Range',
'DCI',   ':CURR:DC:Range',
'ACV',   ':VOLT:AC:Range',
'DCV',   ':VOLT:DC:Range',
'2OHM',  ':RES:Range',
'4OHM',  ':FRES:Range',
'FREQ',  ':FREQ:THR:VOLT:Range',
);

%::dmm_range_values = (
$::dmm_range_functions{'ACI'},   ',1,3,AUTO,',
$::dmm_range_functions{'DCI'},   ',0.01,0.1,1,3,AUTO,',
$::dmm_range_functions{'ACV'},   ',0.1,1,10,100,750,AUTO,',
$::dmm_range_functions{'DCV'},   ',0.1,1,10,100,1000,AUTO,',
$::dmm_range_functions{'2OHM'},  ',0.1,1,10,100,1000,10000,100000,AUTO,',
$::dmm_range_functions{'4OHM'},  ',0.1,1,10,100,1000,10000,100000,AUTO,',
$::dmm_range_functions{'FREQ'},  ',0.1,1,10,100,1000,'
);

%::dmm_unit_functions = (
'ACV',   ':UNIT:VOLT:AC',
'DCV',   ':UNIT:VOLT:DC',
'TEMP',  ':UNIT:TEMP',
);

%::dmm_unit_values = (
$::dmm_unit_functions{'ACV'},   ',V,DB,DBM,',
$::dmm_unit_functions{'DCV'},   ',V,DB,DBM,',
$::dmm_unit_functions{'TEMP'},  ',C,CEL,F,FAR,K,'
);

}elsif(
$::NodeName eq 'm2000drv'
){###############################################################################

#STARS server
$::Server = 'localhost';
#$::Server = '192.168.1.201';

## For NPORT interface
$::NPORT_HOST  = '192.168.11.123'; #NPort host name.
$::NPORT_PORT  = 4002;             #NPort port number.

#$::DESTINATION{'DMM'} = "1";
$::DESTINATION{'ch01'} = "1";
$::DESTINATION{'ch02'} = "2";
$::DESTINATION{'ch03'} = "3";
$::DESTINATION{'ch04'} = "4";
$::ALIAS{'all'} = "ch01,ch02,ch03,ch04";
#$::ALIAS{'all'} = "DMM";
$::NPLC_MAX = 50;

$::RegFile = 'reg.txt';		#Registry fileM

$::DMMTYPE_M2000 = 1;

#functions----------------------------------------
%::dmm_func_functions = (
'ACI',   'CURR:AC',
'DCI',   'CURR:DC',
'ACV',   'VOLT:AC',
'DCV',   'VOLT:DC',
'2OHM',  'RES',
'4OHM',  'FRES',
'FREQ',  'FREQ',
'TEMP',  'TEMP'
);

%::dmm_NPLC_functions = (
'ACI',   ':CURR:AC:NPLC',
'DCI',   ':CURR:DC:NPLC',
'ACV',   ':VOLT:AC:NPLC',
'DCV',   ':VOLT:DC:NPLC',
'2OHM',  ':RES:NPLC',
'4OHM',  ':FRES:NPLC',
'TEMP',  ':TEMP:NPLC',
);

%::dmm_range_functions = (
'ACI',   ':CURR:AC:Range',
'DCI',   ':CURR:DC:Range',
'ACV',   ':VOLT:AC:Range',
'DCV',   ':VOLT:DC:Range',
'2OHM',  ':RES:Range',
'4OHM',  ':FRES:Range',
'FREQ',  ':FREQ:THR:VOLT:Range',
);

%::dmm_range_values = (
$::dmm_range_functions{'ACI'},   ',1,3,AUTO,',
$::dmm_range_functions{'DCI'},   ',0.01,0.1,1,3,AUTO,',
$::dmm_range_functions{'ACV'},   ',0.1,1,10,100,750,AUTO,',
$::dmm_range_functions{'DCV'},   ',0.1,1,10,100,1000,AUTO,',
$::dmm_range_functions{'2OHM'},  ',0.1,1,10,100,1000,10000,100000,AUTO,',
$::dmm_range_functions{'4OHM'},  ',0.1,1,10,100,1000,10000,100000,AUTO,',
$::dmm_range_functions{'FREQ'},  ',0.1,1,10,100,1000,'
);

%::dmm_unit_functions = (
'ACV',   ':UNIT:VOLT:AC',
'DCV',   ':UNIT:VOLT:DC',
'TEMP',  ':UNIT:TEMP',
);

%::dmm_unit_values = (
$::dmm_unit_functions{'ACV'},   ',V,DB,DBM,',
$::dmm_unit_functions{'DCV'},   ',V,DB,DBM,',
$::dmm_unit_functions{'TEMP'},  ',C,CEL,F,FAR,K,'
);
################################################################################

}else{
	die "Bad node name.";
}
1;


m2701drv/configfiles/testconfig1.txt
m2701drv:DMM.DCV.Range=AUTO
m2701drv:CH101.DCV.Range=AUTO
m2701drv:DMM.Function=DCV
m2701drv:CH101.Function=ACV
m2701drv:DMM.DCV.AverageCount=10
m2701drv:CH101.ACV.Range=AUTO
m2701drv:CH111.Function=DCV
m2701drv:CH102.Function=DCV
m2701drv:CH103.Function=DCV
m2701drv:CH104.Function=DCV
m2701drv:setting.CARD1=VOLT
m2701drv:setting.CARD1.CH101=VOLT


m2701drv/m2701drv/config.pl
#Configuration of m2xxxdrv Client.

if(
$::NodeName eq 'm2701drv'
){###############################################################################

#STARS server
$::Server = 'localhost';

## For NPORT interface
$::NPORT_HOST  = '192.168.156.51'; #NPort host name.
$::NPORT_PORT  = 1394;             #NPort port number.

$::ALIAS{'all'} = "DMM";
$::DESTINATION{'DMM'} = "DMM";

$::NPLC_MAX = 50;

$::RegFile = 'reg.txt';		#Registry fileM

$::DMMTYPE_M2000 = 0;

#functions----------------------------------------
%::dmm_func_functions = (
'ACI',   'CURR:AC',
'DCI',   'CURR:DC',
'ACV',   'VOLT:AC',
'DCV',   'VOLT:DC',
'2OHM',  'RES',
'4OHM',  'FRES',
'FREQ',  'FREQ',
'TEMP',  'TEMP'
);

%::dmm_NPLC_functions = (
'ACI',   ':CURR:AC:NPLC',
'DCI',   ':CURR:DC:NPLC',
'ACV',   ':VOLT:AC:NPLC',
'DCV',   ':VOLT:DC:NPLC',
'2OHM',  ':RES:NPLC',
'4OHM',  ':FRES:NPLC',
'TEMP',  ':TEMP:NPLC',
);

%::dmm_range_functions = (
'ACI',   ':CURR:AC:Range',
'DCI',   ':CURR:DC:Range',
'ACV',   ':VOLT:AC:Range',
'DCV',   ':VOLT:DC:Range',
'2OHM',  ':RES:Range',
'4OHM',  ':FRES:Range',
'FREQ',  ':FREQ:THR:VOLT:Range',
);

%::dmm_range_values = (
$::dmm_range_functions{'ACI'},   ',1,3,AUTO,',
$::dmm_range_functions{'DCI'},   ',0.01,0.1,1,3,AUTO,',
$::dmm_range_functions{'ACV'},   ',0.1,1,10,100,750,AUTO,',
$::dmm_range_functions{'DCV'},   ',0.1,1,10,100,1000,AUTO,',
$::dmm_range_functions{'2OHM'},  ',0.1,1,10,100,1000,10000,100000,AUTO,',
$::dmm_range_functions{'4OHM'},  ',0.1,1,10,100,1000,10000,100000,AUTO,',
$::dmm_range_functions{'FREQ'},  ',0.1,1,10,100,1000,'
);

%::dmm_unit_functions = (
'ACV',   ':UNIT:VOLT:AC',
'DCV',   ':UNIT:VOLT:DC',
'TEMP',  ':UNIT:TEMP',
);

%::dmm_unit_values = (
$::dmm_unit_functions{'ACV'},   ',V,DB,DBM,',
$::dmm_unit_functions{'DCV'},   ',V,DB,DBM,',
$::dmm_unit_functions{'TEMP'},  ',C,CEL,F,FAR,K,'
);

%::dmm_avgtop_functions = (
'ACI',   'CURR:AC:AVER',
'DCI',   'CURR:DC:AVER',
'ACV',   'VOLT:AC:AVER',
'DCV',   'VOLT:DC:AVER',
'2OHM',  'RES:AVER',
'4OHM',  'FRES:AVER',
'FREQ',  'FREQ:AVER',
'TEMP',  'TEMP:AVER'
);
}
elsif(
$::NodeName eq 'm2701drvlog'
){###############################################################################

#STARS server
$::Server = 'localhost';

## For NPORT interface
$::NPORT_HOST  = '192.168.156.52'; #NPort host name.
$::NPORT_PORT  = 1394;             #NPort port number.

$::DESTINATION{'ch01'} = "101";
$::DESTINATION{'ch02'} = "102";
$::DESTINATION{'ch03'} = "103";
$::DESTINATION{'ch04'} = "104";
$::DESTINATION{'ch05'} = "105";
$::DESTINATION{'ch06'} = "106";
$::DESTINATION{'ch07'} = "107";
$::DESTINATION{'ch08'} = "108";
$::DESTINATION{'ch09'} = "109";
$::DESTINATION{'ch10'} = "110";
$::DESTINATION{'ch11'} = "111";
$::DESTINATION{'ch12'} = "112";
$::DESTINATION{'ch13'} = "113";
$::DESTINATION{'ch14'} = "114";
$::ALIAS{'all'} = "ch01,ch02,ch03,ch04,ch05,ch06,ch07,ch08,ch09,ch10,ch11,ch12,ch13,ch14";


$::NPLC_MAX = 50;

$::RegFile = 'reg.txt';		#Registry fileM

$::DMMTYPE_M2000 = 0;

#functions----------------------------------------
%::dmm_func_functions = (
'ACI',   'CURR:AC',
'DCI',   'CURR:DC',
'ACV',   'VOLT:AC',
'DCV',   'VOLT:DC',
'2OHM',  'RES',
'4OHM',  'FRES',
'FREQ',  'FREQ',
'TEMP',  'TEMP'
);

%::dmm_NPLC_functions = (
'ACI',   ':CURR:AC:NPLC',
'DCI',   ':CURR:DC:NPLC',
'ACV',   ':VOLT:AC:NPLC',
'DCV',   ':VOLT:DC:NPLC',
'2OHM',  ':RES:NPLC',
'4OHM',  ':FRES:NPLC',
'TEMP',  ':TEMP:NPLC',
);

%::dmm_range_functions = (
'ACI',   ':CURR:AC:Range',
'DCI',   ':CURR:DC:Range',
'ACV',   ':VOLT:AC:Range',
'DCV',   ':VOLT:DC:Range',
'2OHM',  ':RES:Range',
'4OHM',  ':FRES:Range',
'FREQ',  ':FREQ:THR:VOLT:Range',
);

%::dmm_range_values = (
$::dmm_range_functions{'ACI'},   ',1,3,AUTO,',
$::dmm_range_functions{'DCI'},   ',0.01,0.1,1,3,AUTO,',
$::dmm_range_functions{'ACV'},   ',0.1,1,10,100,750,AUTO,',
$::dmm_range_functions{'DCV'},   ',0.1,1,10,100,1000,AUTO,',
$::dmm_range_functions{'2OHM'},  ',0.1,1,10,100,1000,10000,100000,AUTO,',
$::dmm_range_functions{'4OHM'},  ',0.1,1,10,100,1000,10000,100000,AUTO,',
$::dmm_range_functions{'FREQ'},  ',0.1,1,10,100,1000,'
);

%::dmm_unit_functions = (
'ACV',   ':UNIT:VOLT:AC',
'DCV',   ':UNIT:VOLT:DC',
'TEMP',  ':UNIT:TEMP',
);

%::dmm_unit_values = (
$::dmm_unit_functions{'ACV'},   ',V,DB,DBM,',
$::dmm_unit_functions{'DCV'},   ',V,DB,DBM,',
$::dmm_unit_functions{'TEMP'},  ',C,CEL,F,FAR,K,'
);

%::dmm_avgtop_functions = (
'ACI',   'CURR:AC:AVER',
'DCI',   'CURR:DC:AVER',
'ACV',   'VOLT:AC:AVER',
'DCV',   'VOLT:DC:AVER',
'2OHM',  'RES:AVER',
'4OHM',  'FRES:AVER',
'FREQ',  'FREQ:AVER',
'TEMP',  'TEMP:AVER'
);

}elsif(
$::NodeName eq 'm2000drv'
){###############################################################################

#STARS server
$::Server = 'localhost';
#$::Server = '192.168.1.201';

## For NPORT interface
$::NPORT_HOST  = '192.168.11.123'; #NPort host name.
$::NPORT_PORT  = 4002;             #NPort port number.

#$::DESTINATION{'DMM'} = "1";
$::DESTINATION{'ch01'} = "1";
$::DESTINATION{'ch02'} = "2";
$::DESTINATION{'ch03'} = "3";
$::DESTINATION{'ch04'} = "4";
$::ALIAS{'all'} = "ch01,ch02,ch03,ch04";
#$::ALIAS{'all'} = "DMM";
$::NPLC_MAX = 50;

$::RegFile = 'reg.txt';		#Registry fileM

$::DMMTYPE_M2000 = 1;

#functions----------------------------------------
%::dmm_func_functions = (
'ACI',   'CURR:AC',
'DCI',   'CURR:DC',
'ACV',   'VOLT:AC',
'DCV',   'VOLT:DC',
'2OHM',  'RES',
'4OHM',  'FRES',
'FREQ',  'FREQ',
'TEMP',  'TEMP'
);

%::dmm_NPLC_functions = (
'ACI',   ':CURR:AC:NPLC',
'DCI',   ':CURR:DC:NPLC',
'ACV',   ':VOLT:AC:NPLC',
'DCV',   ':VOLT:DC:NPLC',
'2OHM',  ':RES:NPLC',
'4OHM',  ':FRES:NPLC',
'TEMP',  ':TEMP:NPLC',
);

%::dmm_range_functions = (
'ACI',   ':CURR:AC:Range',
'DCI',   ':CURR:DC:Range',
'ACV',   ':VOLT:AC:Range',
'DCV',   ':VOLT:DC:Range',
'2OHM',  ':RES:Range',
'4OHM',  ':FRES:Range',
'FREQ',  ':FREQ:THR:VOLT:Range',
);

%::dmm_range_values = (
$::dmm_range_functions{'ACI'},   ',1,3,AUTO,',
$::dmm_range_functions{'DCI'},   ',0.01,0.1,1,3,AUTO,',
$::dmm_range_functions{'ACV'},   ',0.1,1,10,100,750,AUTO,',
$::dmm_range_functions{'DCV'},   ',0.1,1,10,100,1000,AUTO,',
$::dmm_range_functions{'2OHM'},  ',0.1,1,10,100,1000,10000,100000,AUTO,',
$::dmm_range_functions{'4OHM'},  ',0.1,1,10,100,1000,10000,100000,AUTO,',
$::dmm_range_functions{'FREQ'},  ',0.1,1,10,100,1000,'
);

%::dmm_unit_functions = (
'ACV',   ':UNIT:VOLT:AC',
'DCV',   ':UNIT:VOLT:DC',
'TEMP',  ':UNIT:TEMP',
);

%::dmm_unit_values = (
$::dmm_unit_functions{'ACV'},   ',V,DB,DBM,',
$::dmm_unit_functions{'DCV'},   ',V,DB,DBM,',
$::dmm_unit_functions{'TEMP'},  ',C,CEL,F,FAR,K,'
);

%::dmm_avgtop_functions = (
'ACI',   'CURR:AC:AVER',
'DCI',   'CURR:DC:AVER',
'ACV',   'VOLT:AC:AVER',
'DCV',   'VOLT:DC:AVER',
'2OHM',  'RES:AVER',
'4OHM',  'FRES:AVER',
'FREQ',  'FREQ:AVER',
'TEMP',  'TEMP:AVER'
);
################################################################################

}else{
	die "Bad node name.";
}
1;
